src/review-form.ts

```diff
@@ -0,0 +1,255 @@
+
+  /* The host for archive endpoints and data */
+  @property({ type: String }) baseHost: string = 'https://archive.org';
+
+  /* The path for the endpoint we're submitting to */
+  @property({ type: String }) endpointPath: string = '/write-review.php';
+
+  /* The previous review to pre-fill, if any */
+  @property({ type: Object }) oldReview?: Review;
+
+  /* Errors to add to the form, if any */
+  @property({ type: Array }) errors?: string[];
+
+  /* Number of stars currently selected */
+  @state()
+  private currentStars: number = 0;
+
+  render() {
+    return html`<form action=${`${this.baseHost}${this.endpointPath}`}>
+      ${this.errors
+        ? html`<div class="errors">${this.errors.join(' ')}</div>`
+        : nothing}
+      ${this.starsInputTemplate} ${this.subjectInputTemplate}
+      ${this.bodyInputTemplate}
+      ${this.identifier
+        ? html`<input
+            type="hidden"
+            name="identifier"
+            .value=${this.identifier}
+          />`
+        : nothing}
+      ${this.token
+        ? html`<input
+            type="hidden"
+            name="field_reviewtoken"
+            .value=${this.token}
+          />`
+        : nothing}
+      ${this.actionButtonsTemplate}
+    </form>`;
+  }
+
+  protected updated(changed: PropertyValues): void {
+    if (changed.has('oldReview') && this.oldReview) {
+      this.currentStars = this.oldReview.stars;
+    }
+  }
+
+  private get starsInputTemplate(): HTMLTemplateResult {
+    return html`
+      <div class="form-heading">
+        <label for="stars-field">Rating (optional)</label>
+      </div>
+      <input
+        type="hidden"
+        name="field_stars"
+        id="stars-input"
+        .value=${this.currentStars}
+        required
+      />
+      <div class="stars">
+        ${[1, 2, 3, 4, 5].map(num => this.renderStar(num))}
+        <button class="clear-stars-btn" @click=${this.handleClearBtnClicked}>
+          Clear
+        </button>
+      </div>
+    `;
+  }
+
+  private get subjectInputTemplate(): HTMLTemplateResult {
+    return html`<div class="form-heading">
+        <label for="subject-input">Subject</label>
+      </div>
+      <input
+        type="text"
+        name="field_reviewtitle"
+        id="subject-input"
+        .value=${this.oldReview?.subject ?? ''}
+        required
+      />`;
+  }
+
+  private get bodyInputTemplate(): HTMLTemplateResult {
+    return html`<div class="form-heading">
+        <label for="body-input">Review</label>
+      </div>
+      <textarea
+        name="field_reviewbody"
+        id="body-input"
+        .value=${this.oldReview?.body ?? ''}
+        rows="10"
+        cols="50"
+        required
+      ></textarea>`;
+  }
+
+  private get actionButtonsTemplate(): HTMLTemplateResult {
+    return html`<div class="action-btns">
+      ${this.identifier
+        ? html`<a
+            class="ia-button dark"
+            href=${`${this.baseHost}/details/${this.identifier}`}
```

Comment:
Ah I think this line is a good reason to have the baseHost separate from the endpoint!